lib/mathOps.js

```diff
@@ -263,30 +263,32 @@ net.propagate = function (xIn, layers) {
 };
 
 net.sumAndNormalize = function(youts, isRegression) {
-  var ysums = [], index, row, len, sumRow, index2, len2, outDist = [],
-    isRegression, ysum, rowSum, index3, len3, newRow, yout, sum, norm;
-  for (index = 0, len = youts[0].length; index < len; index++) {
-    sumRow = [];
-    row = youts[0][index];
-    for (index2 = 0, len2 = row.length; index2 < len2; index2++) {
-      sum = 0.0;
-      for (index3 = 0, len3 = youts.length; index3 < len3; index3++) {
-        yout = youts[index3];
-        sum += yout[index][index2];
-      }
-      sumRow.push(sum);
-    }
-    ysums.push(sumRow);
-  }
 
-  outDist = [];
   if (isRegression) {
-    for (index = 0, len = ysums.length; index < len; index++) {
-      ysum = ysums[index];
-      norm = ysums.length;
-      outDist.push([ysum[0] / norm]);
-    }
+    var norm = youts.length;
```

Comment:
we usually use the jslint convention that expects all variables to be defined at the beginning of the function.

Reply:
I'll change that myself in next commit. Don't worry about it.